random/seeded_random.ts

```diff
@@ -0,0 +1,301 @@
+  | 0x70 | 0x71 | 0x72 | 0x73 | 0x74 | 0x75 | 0x76 | 0x77 | 0x78 | 0x79 | 0x7a | 0x7b | 0x7c | 0x7d | 0x7e | 0x7f
+  | 0x80 | 0x81 | 0x82 | 0x83 | 0x84 | 0x85 | 0x86 | 0x87 | 0x88 | 0x89 | 0x8a | 0x8b | 0x8c | 0x8d | 0x8e | 0x8f
+  | 0x90 | 0x91 | 0x92 | 0x93 | 0x94 | 0x95 | 0x96 | 0x97 | 0x98 | 0x99 | 0x9a | 0x9b | 0x9c | 0x9d | 0x9e | 0x9f
+  | 0xa0 | 0xa1 | 0xa2 | 0xa3 | 0xa4 | 0xa5 | 0xa6 | 0xa7 | 0xa8 | 0xa9 | 0xaa | 0xab | 0xac | 0xad | 0xae | 0xaf
+  | 0xb0 | 0xb1 | 0xb2 | 0xb3 | 0xb4 | 0xb5 | 0xb6 | 0xb7 | 0xb8 | 0xb9 | 0xba | 0xbb | 0xbc | 0xbd | 0xbe | 0xbf
+  | 0xc0 | 0xc1 | 0xc2 | 0xc3 | 0xc4 | 0xc5 | 0xc6 | 0xc7 | 0xc8 | 0xc9 | 0xca | 0xcb | 0xcc | 0xcd | 0xce | 0xcf
+  | 0xd0 | 0xd1 | 0xd2 | 0xd3 | 0xd4 | 0xd5 | 0xd6 | 0xd7 | 0xd8 | 0xd9 | 0xda | 0xdb | 0xdc | 0xdd | 0xde | 0xdf
+  | 0xe0 | 0xe1 | 0xe2 | 0xe3 | 0xe4 | 0xe5 | 0xe6 | 0xe7 | 0xe8 | 0xe9 | 0xea | 0xeb | 0xec | 0xed | 0xee | 0xef
+  | 0xf0 | 0xf1 | 0xf2 | 0xf3 | 0xf4 | 0xf5 | 0xf6 | 0xf7 | 0xf8 | 0xf9 | 0xfa | 0xfb | 0xfc | 0xfd | 0xfe | 0xff;
+
+/**
+ * Resumable state for a {@linkcode SeededRandom}.
+ */
+export type SeededRandomState = {
+  /**
+   * The algorithm. Currently, only `pcg32` is supported.
+   */
+  algorithm: "pcg32";
+  /**
+   * The current changeable state of the PCG random number generator.
+   * This value changes upon each iteration of the random number generator.
+   */
+  state: readonly Byte[] & { length: 8 };
+  /**
+   * The amount used to increment each iteration of the PCG random number generator.
+   * This value affects the random numbers generated but remains constant.
+   */
+  inc: readonly Byte[] & { length: 8 };
+};
+
+/**
+ * A seeded pseudo-random number generator, implementing the
+ * {@link https://en.wikipedia.org/wiki/Permuted_congruential_generator | PCG32}
+ * algorithm.
+ *
+ * @example Usage
+ * ```ts
+ * import { SeededRandom } from "@std/random";
+ * import { assertEquals } from "@std/assert";
+ *
+ * const prng = new SeededRandom({
+ *  seed: [
+ *    0x19, 0x57, 0xc5, 0x20, 0x6d, 0x56, 0xb3, 0xf4,
+ *    0x6e, 0xa1, 0x3c, 0x00, 0xb4, 0x6b, 0xcf, 0xa9,
+ *  ],
+ * });
+ * assertEquals(prng.random(), 0.9590322358999401);
+ * assertEquals(prng.random(), 0.3763687589671463);
+ * assertEquals(prng.random(), 0.5337617760524154);
+ * ```
+ */
+export class SeededRandom {
+  #state!: bigint;
+  #inc!: bigint;
+
+  /**
+   * Constructs a new instance.
+   *
+   * @param seed - The seed for the random number generator, which can be
+   * either a 2-tuple of bigints or a single bigint, which
+   * will be converted to a state 2-tuple.
+   *
+   * @example Usage
+   * ```ts no-assert
+   * import { SeededRandom } from "@std/random";
+   * const prng = new SeededRandom({
+   *  seed: [
+   *    0x19, 0x57, 0xc5, 0x20, 0x6d, 0x56, 0xb3, 0xf4,
+   *    0x6e, 0xa1, 0x3c, 0x00, 0xb4, 0x6b, 0xcf, 0xa9,
+   *  ],
+   * });
+   * ```
+   */
+  constructor({ seed }: SeededRandomConstructorOptions) {
```

Comment:
`seed` will always be required for this constructor.
```suggestion
  constructor(seed: <type>) {
```